opensrp-reveal/src/test/java/org/smartregister/reveal/presenter/ListTaskPresenterTest.java

```diff
@@ -112,14 +137,172 @@ public void testOnLocationValidatedCallsStartFormWithCorrectArgumentsForMosquito
+        listTaskPresenter.onInterventionFormDetailsFetched(mock(MosquitoCollectionCardDetails.class));
+        Assert.assertTrue(Whitebox.getInternalState(listTaskPresenter, "changeMosquitoCollectionStatus"));
+    }
+
+    @Test
+    public void testOnFeatureSelectedShouldShowErrorDialogWhenTaskIdentifierIsNull() throws Exception {
+        Feature feature = mock(Feature.class);
+
+        PreferencesUtil preferencesUtil = mock(PreferencesUtil.class);
+        Whitebox.setInternalState(listTaskPresenter, "prefsUtil", preferencesUtil);
+        doReturn("").when(preferencesUtil).getCurrentOperationalArea();
+        doNothing().when(listTaskViewSpy).displayNotification(anyString());
+
+        final String CUSTOM_ERROR_MESSAGE = "My error message";
+
+        Context context = mock(Context.class);
+        doReturn(context).when(listTaskViewSpy).getContext();
+        doReturn(CUSTOM_ERROR_MESSAGE).when(context).getString(anyInt(), any());
+
+        Whitebox.invokeMethod(listTaskPresenter, "onFeatureSelected", feature);
+
+        verify(listTaskViewSpy, times(1)).displayNotification(eq(CUSTOM_ERROR_MESSAGE));
+    }
+
+    @Test
+    public void testfetchMosquitoCollectionDetailsIsCalledForCompleteMosquitoCollectionTask() throws Exception {
+        ListTaskInteractor listTaskInteractor = mock(ListTaskInteractor.class);
+        Whitebox.setInternalState(listTaskPresenter, "listTaskInteractor", listTaskInteractor);
+
+        mockStatic(Utils.class);
+
+        doNothing().when(listTaskInteractor).fetchMosquitoCollectionDetails(anyString(), anyBoolean());
+        when(Utils.getPropertyValue(any(Feature.class), eq(TASK_BUSINESS_STATUS))).thenReturn(COMPLETE);
+        when(Utils.getPropertyValue(any(Feature.class), eq(TASK_CODE))).thenReturn(MOSQUITO_COLLECTION);
+
+        Feature feature = mock(Feature.class);
+        doReturn(true).when(feature).hasProperty(TASK_IDENTIFIER);
+
+        Whitebox.invokeMethod(listTaskPresenter, "onFeatureSelected", feature);
+
+        verify(listTaskInteractor, times(1)).fetchMosquitoCollectionDetails(AdditionalMatchers.or(anyString(), isNull()), eq(false));
+    }
+
+    @Test
+    public void testvalidateUserLocationIsCalledForInCompleteMosquitoCollectionTask() throws Exception {
+        ListTaskInteractor listTaskInteractor = mock(ListTaskInteractor.class);
+
+        ListTaskPresenter listTaskPresenter = spy(this.listTaskPresenter);
+
+        Whitebox.setInternalState(listTaskPresenter, "listTaskInteractor", listTaskInteractor);
+
+        mockStatic(Utils.class);
+
+        doNothing().when(listTaskInteractor).fetchMosquitoCollectionDetails(anyString(), anyBoolean());
+        when(Utils.getPropertyValue(any(Feature.class), eq(TASK_BUSINESS_STATUS))).thenReturn(INCOMPLETE);
+        when(Utils.getPropertyValue(any(Feature.class), eq(TASK_CODE))).thenReturn(MOSQUITO_COLLECTION);
+        PowerMockito.doNothing().when(listTaskPresenter, "validateUserLocation");
```

Comment:
Remove this, the object under test shouldn't be disabled from executing methods.  verify the methods on `locationPresenter` and `listTaskView` mocks were invoked instead

Reply:
Done.